pkg/ddevapp/laravel.go

```diff
@@ -18,35 +19,31 @@ func isLaravelApp(app *DdevApp) bool {
 	return fileutil.FileExists(filepath.Join(app.AppRoot, "artisan"))
 }
 
-func envSettingsWarning(status int) {
-	var srcFile = ".env"
-	var message = "Don't forget to configure the database in your .env file"
-
-	if WarnTypeAbsent == status {
-		srcFile += ".example"
-		message = "Don't forget to create the .env file with proper database settings"
-	}
-	util.Warning(message)
-	util.Warning("You can do it with this one-liner:")
-	util.Warning("ddev exec \"cat %v | sed  -E 's/DB_(HOST|DATABASE|USERNAME|PASSWORD)=(.*)/DB_\\1=db/g' > .env\"", srcFile)
-	util.Warning("Read more on https://ddev.readthedocs.io/en/stable/users/quickstart/#laravel")
-}
-
 func laravelPostStartAction(app *DdevApp) error {
-	if fileutil.FileExists(filepath.Join(app.AppRoot, ".env")) {
-		isConfiguredDbHost, err := fileutil.FgrepStringInFile(app.SiteSettingsPath, `DB_HOST=db`)
-		isConfiguredDbConnection, _ := fileutil.FgrepStringInFile(app.SiteSettingsPath, `DB_CONNECTION=ddev`)
-		if err == nil && !isConfiguredDbHost && !isConfiguredDbConnection {
-			envSettingsWarning(WarnTypeNotConfigured)
+	envContents, err := ReadEnvFile(app)
+	if err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("Unable to read .env file: %v", err)
+	}
+	if os.IsNotExist(err) {
+		err = fileutil.CopyFile(filepath.Join(app.AppRoot, ".env.example"), filepath.Join(app.AppRoot, ".env"))
+		if err != nil {
+			return err
+		}
+		envContents, err = ReadEnvFile(app)
+		if err != nil {
+			return err
 		}
-	} else {
-		envSettingsWarning(WarnTypeAbsent)
+	}
+	envContents["DB_HOST"] = "db"
+	envContents["DB_PORT"] = "3306"
```

Comment:
This assumes a "MySql" database type. Doesn't DDEV have a DB_PORT variable we could use instead?

Reply:
I did some quick look-around and didn't see support for postgres on Laravel. Does it have it?

Reply:
It definately has it: https://laravel.com/docs/9.x/database#introduction
My company tends to use Postgres for most of its projects, it was one of the reason I pushed for the ddev-contrib module to get first-party support in DDEV.

See https://github.com/drud/ddev/pull/4197#discussion_r966558313